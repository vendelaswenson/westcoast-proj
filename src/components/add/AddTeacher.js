function AddTeacher() {
    return (
        <>
        <h2>Add teacher</h2>
      <form>
        <label>Teachers first name:
          <input type="text" />
        </label>
        <label>Teachers last name:
          <input type="text" />
        </label>
        <label>Social security number:
          <input type="number" />
        </label>
        <label>E-mail:
          <input type="text" />
        </label>
        <label>Phone number:
          <input type="number" />
        </label>
        <label>List of competencies:
          <input type="text" />
        </label>
        <button>Add</button>
      </form>
      </>
    )
  }

  export default AddTeacher