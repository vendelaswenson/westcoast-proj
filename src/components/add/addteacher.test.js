import AddTeacher from './AddTeacher';
import {render, cleanup, screen} from '@testing-library/react'
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

afterEach(() => {
    cleanup(); 
})

describe("Component", () => {
  it("renders without crashing", async () => {
    const utils = render(<AddTeacher />, {wrapper: MemoryRouter});
    expect(utils).toMatchSnapshot();
    expect(utils).toBeTruthy();
  });

  it("Length label should be in the document", () => {
    render(<AddTeacher />, {wrapper: MemoryRouter});
    expect(screen.getByTestId('add-word-button')).toBeEnabled();
  })
});