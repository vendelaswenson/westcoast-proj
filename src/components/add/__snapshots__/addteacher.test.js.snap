// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="teacher-form"
      >
         
        <h2
          class="add-h2"
        >
          Add teacher
        </h2>
        <form>
          <div>
            <label
              for="firstName"
            >
              Teachers first name:  
            </label>
            <input
              id="firstName"
              name="firstName"
              type="text"
            />
          </div>
          <div>
            <label
              for="lastName"
            >
              Teachers last name:  
            </label>
            <input
              id="lastName"
              name="lastName"
              type="text"
            />
          </div>
          <div>
            <label
              for="socSecNum"
            >
              Social security number:  
            </label>
            <input
              id="socSecNum"
              name="socSecNum"
              type="text"
            />
          </div>
          <div>
            <label
              for="email"
            >
              E-mail:  
            </label>
            <input
              id="email"
              name="email"
              type="text"
            />
          </div>
          <div>
            <label
              for="phoneNum"
            >
              Phone number:  
            </label>
            <input
              id="phoneNum"
              name="phoneNum"
              type="text"
            />
          </div>
          <div>
            <label
              for="competence"
            >
              Competencies:  
            </label>
            <input
              id="competence"
              name="competence"
              type="text"
            />
          </div>
          <div
            class="btns-container"
          >
            <button
              class="btn"
            >
              Add
            </button>
            <button
              class="btn"
            >
              Cancel
            </button>
          </div>
        </form>
          
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="teacher-form"
    >
       
      <h2
        class="add-h2"
      >
        Add teacher
      </h2>
      <form>
        <div>
          <label
            for="firstName"
          >
            Teachers first name:  
          </label>
          <input
            id="firstName"
            name="firstName"
            type="text"
          />
        </div>
        <div>
          <label
            for="lastName"
          >
            Teachers last name:  
          </label>
          <input
            id="lastName"
            name="lastName"
            type="text"
          />
        </div>
        <div>
          <label
            for="socSecNum"
          >
            Social security number:  
          </label>
          <input
            id="socSecNum"
            name="socSecNum"
            type="text"
          />
        </div>
        <div>
          <label
            for="email"
          >
            E-mail:  
          </label>
          <input
            id="email"
            name="email"
            type="text"
          />
        </div>
        <div>
          <label
            for="phoneNum"
          >
            Phone number:  
          </label>
          <input
            id="phoneNum"
            name="phoneNum"
            type="text"
          />
        </div>
        <div>
          <label
            for="competence"
          >
            Competencies:  
          </label>
          <input
            id="competence"
            name="competence"
            type="text"
          />
        </div>
        <div
          class="btns-container"
        >
          <button
            class="btn"
          >
            Add
          </button>
          <button
            class="btn"
          >
            Cancel
          </button>
        </div>
      </form>
        
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
